### PR TITLE
[ETHOSN] Improve inferring new shape of the Reshape operator

### PR DESCRIPTION
**Pull request by NicolaLancellotti**
This pr improves the inference of the new shape of the Reshape operator for the Arm(R) Ethos(TM)-N NPU.

cc @lhutton1

### CODE REVIEW

**Comment by NicolaLancellotti** [comment]
@lhutton1 @leandron

File: src/relay/backend/contrib/ethosn/ethosn_api.cc [hunk 1]
```diff
@@ -305,10 +304,15 @@ EthosnError EthosnAPI::Reshape(const Expr& expr, ReshapeParams* params) {
     tensor_size *= dim;
```
Comment (lhutton1) [inline src/relay/backend/contrib/ethosn/ethosn_api.cc:304]: Seems that `tensor_size` doesn't get used? I think we can remove this code

**Review by lhutton1 — COMMENTED** [review]
Thanks for the fix @NicolaLancellotti. Presumably due to this change: https://github.com/apache/tvm/pull/12594/files#diff-0cc1c5de01e0cebda9c447e34492ede3ae3121a9b6e11296218ecae47b299c68L329, the CI is failing on a network test. I also left a small comment below that you might as well add in at the same time :) Otherwise LGTM!

**Comment by lhutton1** [comment]
After reading https://github.com/apache/tvm-rfcs/pull/88 I think we should quickly update the commit body to be more specific about the fix, WDYT?

**Comment by NicolaLancellotti** [comment]
> After reading [apache/tvm-rfcs#88](https://github.com/apache/tvm-rfcs/pull/88) I think we should quickly update the PR body to be more specific about the fix, WDYT? (No need to push a new commit, I can change the commit message so it matches the PR)

It is good to me to update the PR body.

**Review by lhutton1 — APPROVED** [review]
LGTM!

**Comment by lhutton1** [comment]
Thanks @NicolaLancellotti!

**Pull request merged**